libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java

```diff
@@ -0,0 +1,133 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.location.Location;
+import android.os.Handler;
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.geojson.Point;
+import com.mapbox.navigator.NavigationStatus;
+import com.mapbox.services.android.navigation.v5.navigation.camera.SimpleCamera;
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteDetector;
+import com.mapbox.services.android.navigation.v5.route.FasterRouteDetector;
+import com.mapbox.services.android.navigation.v5.snap.SnapToRoute;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.util.Date;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(RobolectricTestRunner.class)
+public class RouteProcessorRunnableTest {
+
+  @Test
+  public void onRun_buildNewRouteProgressReceivesStatusAndRoute() {
+    NavigationRouteProcessor processor = mock(NavigationRouteProcessor.class);
+    NavigationStatus status = buildMockStatus();
+    DirectionsRoute route = mock(DirectionsRoute.class);
+    RouteProcessorRunnable runnable = buildRouteProcessorRunnableWith(processor, status, route);
+    runnable.updateRawLocation(mock(Location.class));
+
+    runnable.run();
+
+    verify(processor).buildNewRouteProgress(status, route);
```

Comment:
Maybe also verify `routeProcessor.updatePreviousRouteProgress` is called as well?

Reply:
@devotaaabel I added another test for this 👍